metricflow/naming/object_builder_scheme.py

```diff
@@ -0,0 +1,200 @@
+                f"The specified input {repr(input_str)} does not match the input described by the object builder "
+                f"pattern."
+            )
+        try:
+            # TODO: Update when more appropriate parsing libraries are available.
+            call_parameter_sets = PydanticWhereFilter(where_sql_template="{{ " + input_str + " }}").call_parameter_sets
+        except ParseWhereFilterException as e:
+            raise ValueError(f"A spec pattern can't be generated from the input string {repr(input_str)}") from e
+
+        num_parameter_sets = (
+            len(call_parameter_sets.dimension_call_parameter_sets)
+            + len(call_parameter_sets.time_dimension_call_parameter_sets)
+            + len(call_parameter_sets.entity_call_parameter_sets)
+        )
+        if num_parameter_sets != 1:
+            raise ValueError(f"Did not find exactly 1 call parameter set. Got: {num_parameter_sets}")
+
+        for dimension_call_parameter_set in call_parameter_sets.dimension_call_parameter_sets:
+            return DimensionPattern(
+                EntityLinkPatternParameterSet(
+                    element_name=dimension_call_parameter_set.dimension_reference.element_name,
+                    entity_links=dimension_call_parameter_set.entity_path,
+                    time_granularity=None,
+                    date_part=None,
+                    fields_to_compare=(
+                        ParameterSetField.ELEMENT_NAME,
+                        ParameterSetField.ENTITY_LINKS,
+                        ParameterSetField.DATE_PART,
+                    ),
+                )
+            )
+
+        for time_dimension_call_parameter_set in call_parameter_sets.time_dimension_call_parameter_sets:
+            fields_to_compare = [
+                ParameterSetField.ELEMENT_NAME,
+                ParameterSetField.ENTITY_LINKS,
+                ParameterSetField.DATE_PART,
+            ]
+
+            if time_dimension_call_parameter_set.time_granularity is not None:
+                fields_to_compare.append(ParameterSetField.TIME_GRANULARITY)
+
+            return TimeDimensionPattern(
+                EntityLinkPatternParameterSet(
+                    element_name=time_dimension_call_parameter_set.time_dimension_reference.element_name,
+                    entity_links=time_dimension_call_parameter_set.entity_path,
+                    time_granularity=time_dimension_call_parameter_set.time_granularity,
+                    date_part=time_dimension_call_parameter_set.date_part,
+                    fields_to_compare=tuple(fields_to_compare),
+                )
+            )
+
+        for entity_call_parameter_set in call_parameter_sets.entity_call_parameter_sets:
+            return EntityLinkPattern(
+                EntityLinkPatternParameterSet(
+                    element_name=entity_call_parameter_set.entity_reference.element_name,
+                    entity_links=entity_call_parameter_set.entity_path,
```

Comment:
Tangentially related - do we actually allow linked entities, like `user__listing`, in a query? Or is this forward-looking for things we might add, or for using this for internal cases where we may need that entity path resolution in an intermediate query step?

Reply:
Really? TIL. I don't think I've ever seen it.